src/routeGeneration/templateHelpers.ts

```diff
@@ -359,6 +363,95 @@ export class ValidationService {
     return new Buffer(value);
   }
 
+  public validateUnion(name: string, value: any, fieldErrors: FieldErrors, subSchemas: TsoaRoute.PropertySchema[] | undefined, parent = ''): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors = subSchemas.map(subSchema => {
+      const subFieldErrors: FieldErrors = {};
+      this.ValidateParam(subSchema, value, name, subFieldErrors, parent);
+      return subFieldErrors;
+    });
+
+    if (!subFieldErrors.some(subFieldError => Object.keys(subFieldError).length === 0)) {
+      fieldErrors[parent + name] = {
+        message: `Could not match the union against any of the items. Issues: ${subFieldErrors}`,
+        value,
+      };
+      return;
+    }
+
+    return value;
+  }
+
+  public validateIntersection(name: string, value: any, fieldErrors: FieldErrors, subSchemas: TsoaRoute.PropertySchema[] | undefined, parent = ''): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors = subSchemas.map(subSchema => {
+      const subFieldErrors: FieldErrors = {};
+      this.ValidateParam(subSchema, value, name, subFieldErrors, parent);
+      return subFieldErrors;
+    });
+
+    subFieldErrors.forEach(subFieldError => {
+      (Object as any).entries(subFieldError).forEach(([key, value]) => {
```

Comment:
@WoH can you please turn on this TsLint rule? https://palantir.github.io/tslint/rules/no-shadowed-variable/

I ask because it looks like you have a variable named `value` here and then you have `value` as a parameter of `validateIntersection`. For readability purposes, please change the `validateIntersection` property to `dataToValidate` (or something else if you can think of something shorter).

Reply:
@WoH I noticed that this was marked as resolved. Was that by mistake? Shadowing a variable name is an almost universally discouraged practice.

Reply:
Yes, because that's exactly the part to be removed because of #411.